tests/Resources/ti.media.audioplayer.test.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions');
+
+describe('Titanium.Media', function () {
+	it('#createAudioPlayer()', function () {
+		should(Ti.Media.createAudioPlayer).be.a.Function;
+	});
+});
+
+describe('Titanium.Media.AudioPlayer', function () {
+	var audioPlayer;
+
+	beforeEach(function () {
+		audioPlayer = Ti.Media.createAudioPlayer({ url: 'sample.mp3' });
+	});
+
+	afterEach(function () {
+		audioPlayer = null;
+	});
+
+	it('apiName', function () {
+		var player = Ti.Media.createAudioPlayer({ url: 'sample.mp3' });
+		should(player).have.a.readOnlyProperty('apiName').which.is.a.String;
+		should(player.apiName).be.eql('Ti.Media.AudioPlayer');
+		player = null;
+	});
+
+	// constants
+	it('STATE_BUFFERING', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_BUFFERING').which.is.a.Number;
+	});
+
+	it('STATE_INITIALIZED', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_INITIALIZED').which.is.a.Number;
+	});
+
+	it('STATE_PAUSED', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_PAUSED').which.is.a.Number;
+	});
+
+	it('STATE_PLAYING', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_PLAYING').which.is.a.Number;
+	});
+
+	it('STATE_STARTING', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_STARTING').which.is.a.Number;
+	});
+
+	it('STATE_STOPPED', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_STOPPED').which.is.a.Number;
+	});
+
+	it('STATE_STOPPING', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_STOPPING').which.is.a.Number;
+	});
+
+	it('STATE_WAITING_FOR_DATA', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_WAITING_FOR_DATA').which.is.a.Number;
+	});
+
+	it('STATE_WAITING_FOR_QUEUE', function () {
+		should(Ti.Media).have.constant('AUDIO_STATE_WAITING_FOR_QUEUE').which.is.a.Number;
+	});
+
+	it('.url', function () {
+		should(audioPlayer.url).be.a.String;
+		should(audioPlayer.getUrl).be.a.Function;
+		should(audioPlayer.setUrl).be.a.Function;
+		should(audioPlayer.url).eql(audioPlayer.getUrl());
+	});
+
+	it('#start', function () {
```

Comment:
Write some more complex test cases, like say this one:

- make sure the audio's `isPlaying()`, `isPaused()`, `playing`, `paused` return false; `idle` returns true. `progress == 0`.
- Call `#start()`
- make sure the audio's `isPlaying()` and `playing` return true, `idle`, `paused` and `isPaused()` return false. Check `state`
- Call `#pause()`
- check the properties/methods again. Check that `progress` > 0.
- Call `#start()`, check the state properties/methods
- Call `#stop()`, check the state properties/methods